src/components/TodoItem.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react/sort-comp */
+import React, { Component } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import PropTypes from 'prop-types';
+import styles from './TodoItem.module.css';
+
+export default class TodoItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editing: false,
+    };
+  }
+
+  handleEditing = () => {
+    this.setState({
+      editing: true,
+    });
+  };
+
+  // editTodo
+  handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      this.setState({ editing: false });
+    }
+  };
+
+  componentWillUnmount() {
+    console.log('Cleaning up...');
+  }
+
+  render() {
+    const {
+      todo, handleChangeProps, deleteTodoProps, setUpdate,
+    } = this.props;
+    const { editing } = this.state;
+
+    const completedStyle = {
+      fontStyle: 'italic',
+      color: '#595959',
+      opacity: 0.4,
+      textDecoration: 'line-through',
+    };
+
+    // delete
+    const viewMode = {};
+    const editMode = {};
+
+    if (editing) {
+      viewMode.display = 'none';
+    } else {
+      editMode.display = 'none';
+    }
+
+    // destruction
+    // const { completed, id, title } = todo;
```

Comment:
_[OPTIONAL]_ to keep your code clean, kindly remove all unused commented-out code before merging your PR